### DISCUSSION TITLE
does otbr-agent support getting the prefix from the dhcpv6 server?

### DISCUSSION THREAD

**Discussion by ssenthu** [openthread/openthread#8744] [Q&A] [answered]
I have a open-wrt router in the network and OTBR does send the router solicitation message but never request a prefix for the Thread nodes from the dhcpv6 prefix-delegation server (Open-wrt router with odhcpd).  Is that feature supported or will be supported in the near future?  

When i assign a prefix manually, that works or OTBR randomly pick a prefix and send the router advertisement. 

\ot-br-posix\third_party\openthread\repo\src\core\net\dhcp6.hpp, are they really used? or do i need to configure my dhcpv6 server bit differently?
/**
 * DHCPv6 Message Types
 *
 */
enum Type : uint8_t
{
    kTypeNone               = 0,
    kTypeSolicit            = 1,
    kTypeAdvertise          = 2,
    kTypeRequest            = 3,
    kTypeConfirm            = 4,
    kTypeRenew              = 5,
    kTypeRebind             = 6,
    kTypeReply              = 7,
    kTypeRelease            = 8,
    kTypeDecline            = 9,
    kTypeReconfigure        = 10,
    kTypeInformationRequest = 11,
    kTypeRelayForward       = 12,
    kTypeRelayReply         = 13,
    kTypeLeaseQuery         = 14,
    kTypeLeaseQueryReply    = 15,
};
@jwhui

**Comment by jwhui** [comment, marked as answer]
While OpenThread does not currently integrate with DHCPv6 Prefix Delegation, we are currently working to support this.

**Reply by ssenthu** [reply]
Thanks @jwhui  for the confirmation.
